cmd/subnetcmd/deploy.go

```diff
@@ -432,22 +411,46 @@ func deploySubnet(cmd *cobra.Command, args []string) error {
 		}
 	}
 
+	fee := network.GenesisParams().CreateBlockchainTxFee
+	if createSubnet {
+		fee += network.GenesisParams().CreateSubnetTxFee
+	}
+	kc, err := keychain.GetKeychainFromCmdLineFlags(
+		app,
+		constants.PayTxsFeesMsg,
+		network,
+		keyName,
+		useEwoq,
+		useLedger,
+		ledgerAddresses,
+		fee,
+	)
+	if err != nil {
+		return err
+	}
+
+	// used in E2E to simulate public network execution paths on a local network
+	if os.Getenv(constants.SimulatePublicNetwork) != "" {
+		network = models.LocalNetwork
+	}
+
 	if createSubnet {
 		// accept only one control keys specification
 		if len(controlKeys) > 0 && sameControlKey {
 			return errMutuallyExlusiveControlKeys
 		}
-		// use creation key as control key
+		// use first fee-paying key as control key
 		if sameControlKey {
-			controlKeys, err = loadCreationKeys(network, kc)
+			kcKeys, err := kc.PChainFormattedStrAddresses()
 			if err != nil {
 				return err
 			}
+			controlKeys = kcKeys[:1]
```

Comment:
we need to check if slice is not empty for whatever reason to avoid `slice bounds out of range [:1] with capacity 0`

Reply:
keychain will error is having 0 addresses